docs/docs-components/DocSearch.js

```diff
@@ -48,34 +122,42 @@ export default function DocSearch({ popoverZIndex }: {| popoverZIndex?: Composit
       apiKey: '88c1825a5951ee68c92b4fbf4e85ec7f',
       debug: false, // Set debug to true if you want to keep open and inspect the dropdown
       indexName: 'gestalt',
-      inputSelector: '#algolia-doc-search',
+      inputSelector: `#${INPUT_ID}`,
     });
   }, []);
 
-  useEffect(() => {
-    document.addEventListener('keydown', handleKeyDown, false);
-
-    return () => {
-      document.removeEventListener('keydown', handleKeyDown, false);
-    };
-  }, []);
-
   return (
-    <Tooltip inline text="Pro tip: press '/' to quickly access search" zIndex={popoverZIndex}>
-      <form className="searchbox">
-        <div className="searchbox__wrapper">
-          <input
-            id="algolia-doc-search"
-            className="searchbox__input"
-            type="search"
-            placeholder="Search Gestalt"
-            aria-label="Search Gestalt docs"
-          />
-          <Box position="absolute" top left marginTop={3} marginStart={4}>
-            <Icon icon="search" accessibilityLabel="" size="16" color="default" />
-          </Box>
-        </div>
-      </form>
-    </Tooltip>
+    <Fragment>
+      <Box
+        alignItems="center"
+        display={isMobileSearchExpandedOpen ? 'flex' : 'none'}
+        lgDisplay="flex"
+        flex="shrink"
+        marginStart={2}
+        mdMarginStart={0}
+      >
+        <Box flex="grow" paddingX={2}>
+          <SearchBox popoverZIndex={PAGE_HEADER_POPOVER_ZINDEX} />
+        </Box>
+      </Box>
+
+      <Box display="block" lgDisplay="none" marginStart={2}>
```

Comment:
I migrated the compressed UI into this file so that all search is together. This component is still very tightly-coupled to Header, though I think that's fine. This is mostly about code organization, not creating independently-reusable components.